javalib/src/main/scala/java/time/Year.scala

```diff
@@ -0,0 +1,203 @@
+    case CENTURIES => plusYears(MathJDK8Bridge.multiplyExact(amount, 100))
+    case MILLENNIA => plusYears(MathJDK8Bridge.multiplyExact(amount, 1000))
+    case ERAS      =>
+      val era = getLong(ERA)
+      `with`(ERA, MathJDK8Bridge.addExact(era, amount))
+
+    case _: ChronoUnit =>
+      throw new UnsupportedTemporalTypeException(s"Unsupported unit: $unit")
+
+    case _ =>
+      unit.addTo(this, amount)
+  }
+
+  def plusYears(amount: Long): Year = {
+    if (amount == 0) {
+      this
+    } else {
+      val newYear = MathJDK8Bridge.addExact(year, amount)
+      Year.of(YEAR.checkValidIntValue(newYear))
+    }
+  }
+
+  override def minus(amount: Long, unit: TemporalUnit): Year = {
+    if (amount != Long.MinValue) plus(-amount, unit)
+    else plus(Long.MaxValue, unit).plus(1, unit)
+  }
+
+  def minusYears(amount: Long): Year = minus(amount, YEARS)
+
+  // Not implemented
+  // def query[R](query: TemporalQuery[R]): R
+
+  def adjustInto(temporal: Temporal): Temporal =
+    temporal.`with`(YEAR, year)
+
+  def until(end: Temporal, unit: TemporalUnit): Long = {
+    val other = Year.from(end)
+    val yearsDiff: Long = other.getValue.toLong - year
+
+    unit match {
+      case YEARS     => yearsDiff
+      case DECADES   => yearsDiff / 10
+      case CENTURIES => yearsDiff / 100
+      case MILLENNIA => yearsDiff / 1000
+      case ERAS      => other.getLong(ERA) - getLong(ERA)
+
+      case _: ChronoUnit =>
+        throw new UnsupportedTemporalTypeException(s"Unsupported unit: $unit")
+
+      case _ => unit.between(this, other)
+    }
+  }
+
+  def atDay(dayOfYear: Int): LocalDate =
+    LocalDate.ofYearDay(year, dayOfYear)
+
+  // TODO
+  // def atMonth(month: Month): YearMonth = YearMonth.of(year, month)
+
+  // TODO
+  // def atMonth(month: Int): YearMonth = YearMonth.of(year, month)
+
+  def atMonthDay(monthDay: MonthDay): LocalDate = monthDay.atYear(year)
+
+  def compareTo(other: Year): Int = year - other.getValue
+
+  def isAfter(other: Year): Boolean = compareTo(other) > 0
+
+  def isBefore(other: Year): Boolean = compareTo(other) < 0
+
+  override def equals(other: Any): Boolean = other match {
+    case that: Year => year == that.getValue
+    case _ => false
+  }
+
+  override def hashCode(): Int = year.hashCode()
+
+  override def toString: String = s"$year"
+
+  // Not implemented
+  // def format(formatter: DateTimeFormatter): String
+
+}
+
+object Year {
+  import ChronoField._
+
+  val MIN_VALUE: Int = -999999999
+  val MAX_VALUE: Int = 999999999
+
+  private val iso = IsoChronology.INSTANCE
+
+  def now(): Year = from(LocalDate.now())
+
+  // Not implemented
+  // def now(zone: ZoneId): Year
+  // def now(clock: Clock): Year
+
+  def isLeap(year: Long): Boolean = iso.isLeapYear(year)
+
+  def of(year: Int): Year = {
+    YEAR.checkValidIntValue(year)
+    new Year(year)
+  }
+
+  def from(temporal: TemporalAccessor): Year = temporal match {
+    case t: Year => t
```

Comment:
Use `case temporal: Year => t`. In general when you just refine the type of a field just rebind it to the same name.